### ISSUE TITLE
onComplete callback not being triggered on complete processing of image.

### ISSUE THREAD

**Issue by AdeelAsif** [scaleflex/filerobot-image-editor#3]
Following is my code snippet:

      <FilerobotImageEditor
        show={show}
        src={src}
        config={config}
        onClose={image => {
          const editedCanvas = document.getElementById(
            'scaleflex-image-edit-box',
          );
          console.log("==>editedCanvas :", editedCanvas, "\n");
        }}
        onComplete={image => {
          console.log('==>On Complete image :', image, '\n');
        }}
      />

Basically I want to be able to set the processed image as source to another image and eventually upload that to my server.

I would really appreciate a quick response as to solve my problem.

**Comment by dzmitry-stramavus** [comment]
@AdeelAsif , hey checking

**Comment by dzmitry-stramavus** [comment]
@AdeelAsif , I added onBeforeComplete method. I think it's what you are looking for.

```onBeforeComplete={(props) => { console.log(props); return true; }}```

where `props: { status, canvas }`

![image](https://user-images.githubusercontent.com/10708539/65319390-36175c00-dba8-11e9-93b9-d60364428b35.png)

to block downloading image you need to return false in the onBeforeComplete method

check the example in code sandbox below

https://codesandbox.io/s/example-onbeforecomplete-handler-vcovn

**available in v3.1.0**
